Configuration.cpp

```diff
@@ -118,7 +118,33 @@ bool Configuration::readConfigFile(char *path) {
 	std::string country = root["countries"]["blacklist"][i].asString();
 
 	trace->traceEvent(TRACE_INFO, "Adding %s to countries blacklist", country.c_str());
-	countries[country2u16((char*)country.c_str())] = MARKER_DROP;
+	count_conts[country2u16((char*)country.c_str())] = MARKER_DROP;
+      }
+    }
+  }
+
+  if(!root["continents"].empty()){
+    if(root["continents"]["whitelist"].empty()){
+      trace->traceEvent(TRACE_INFO, "Missing %s from %s", "continents whitelist", path);
+    }
+    else{
+      for(Json::Value::ArrayIndex i = 0; i != root["continents"]["whitelist"].size(); i++){
+        std::string continent = root["continents"]["whitelist"][i].asString();
+
+        trace->traceEvent(TRACE_INFO, "Adding %s to continents whitelist", continent.c_str());
+        count_conts[country2u16((char*)continent.c_str())] = MARKER_PASS;
+      }
+    }
+
+    if(root["continents"]["blacklist"].empty()){
+      trace->traceEvent(TRACE_INFO, "Missing %s from %s", "continents blacklist", path);
+    }
+    else{
+      for(Json::Value::ArrayIndex i = 0; i != root["continents"]["blacklist"].size(); i++){
+        std::string continent = root["continents"]["blacklist"][i].asString();
+
+        trace->traceEvent(TRACE_INFO, "Adding %s to continents blacklist", continent.c_str());
+        count_conts[country2u16((char*)continent.c_str())] = MARKER_DROP;
```

Comment:
country2u16 is misleading and it needs to be renamed.
Furthermore country2u16() takes as input only the first two chars that might be a problem. Please address this issue